src/Plugin/Field/FieldType/StrawberryFieldItem.php

```diff
@@ -185,6 +185,61 @@ public function isEmpty() {
    }
 
 
+   /**
+    * Decodes main value JSON string into an array
+    *
+    * We don't keep this around because this will be mainly used to be
+    * modified and re encoded afterwards.
+    *
+    * @param bool $assoc
+    *   If return is array or a stdclass Object.
+    *
+    * @return array|\stdClass
+    */
+   public function provideDecoded($assoc = TRUE) {
+     if ($this->isEmpty()) {
+       $this->flattenjson = [];
+       $jsonArray = [];
+     }
+     elseif ($this->validate()->count() == 0) {
+       $mainproperty = $this->mainPropertyName();
+       $jsonArray = json_decode($this->{$mainproperty}, $assoc, 10);
+
+     }
+     return $jsonArray;
+   }
+
+
+   /**
+    * Encodes and sets main value from array
+    *
+    * This method also clears flattenjson and jsonjmesresult caches.
+    *
+    * @param array $jsonarray
+    *   Array of data we want to save in the main property
+    *
+    * @return string|boolean
+    *   Returns either the correctly encoded string or boolean FALSE
+    *   Make sure you compare using === FALSE!.
+    *
+    * @throws  \InvalidArgumentException
+    *    If what is passed to ::setValue() is not an array.
+    */
+   public function setMainValueFromArray(array $jsonarray) {
```

Comment:
::setMainValueFromArray allows me to save an array directly into e.g the field_descriptive_metadata field item, and it will convert to the string and also clear the caches `$this->flattenjson` and `$this->jsonjmesresults`